hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/HoodieSparkParquetReader.java

```diff
@@ -69,7 +69,7 @@ public HoodieSparkParquetReader(HoodieStorage storage, StoragePath path) {
     this.storage = storage.newInstance(path, storage.getConf().newInstance());
     // Avoid adding record in list element when convert parquet schema to avro schema
     this.storage.getConf().set(ADD_LIST_ELEMENT_RECORDS, "false");
-    this.parquetUtils = FileFormatUtils.getInstance(HoodieFileFormat.PARQUET);
+    this.parquetUtils = new HoodieSparkIOFactory(storage).getFileFormatUtils(HoodieFileFormat.PARQUET);
```

Comment:
can't we use `HoodieIOFactory.getIOFactory(storage).getFileFormatUtils(HoodieFileFormat.PARQUET)`?

Reply:
Yes.  Given this Spark so I hardcoded it.  Fix now.